utils/utils.go

```diff
@@ -559,6 +559,15 @@ func InterruptProcessTree(logger *zap.Logger, ppid int, sig syscall.Signal) erro
 	return nil
 }
 
+func SetUmask(mask int) error {
+	prevUmask := syscall.Umask(mask)
```

Comment:
We are still using a linux command to replace for chmod. It would be better if we can use some api or sdk from a linux command to do this. Like now we are using docker sdk instead of docker cli commands for executing docker related operations. Research on it a bit and let me know